lib/queuePopulator/IngestionReader.js

```diff
@@ -1,37 +1,42 @@
-const arsenal = require('arsenal');
 const async = require('async');
 
-const LogConsumer = arsenal.storage.metadata.bucketclient.LogConsumer;
-const BucketClient = require('bucketclient').RESTClient;
 const IngestionProducer = require('./IngestionProducer');
-
 const LogReader = require('./LogReader');
 
 class IngestionReader extends LogReader {
     constructor(params) {
         const { zkClient, kafkaConfig, bucketdConfig, ingestionPath, qpConfig,
-            raftId, logger, extensions, metricsProducer, s3Config } = params;
-        const { host, port, zookeeperSuffix, name } = bucketdConfig;
-        logger.info('initializing raft log reader',
-            { method: 'RaftLogReader.constructor',
-                bucketdConfig, raftId });
-        const bucketClient = new BucketClient(`${host}:${port}`);
-        const logConsumer = new LogConsumer({ bucketClient,
-            raftSession: raftId,
-            logger });
-        super({ zkClient, kafkaConfig, logConsumer, logId: `raft_${raftId}`,
+            logger, extensions, metricsProducer, s3Config, bucket } = params;
+        const { name } = bucketdConfig;
+        super({ zkClient, kafkaConfig, logConsumer: {}, logId: '',
                 logger, extensions, metricsProducer });
-        this.raftId = raftId;
         this._iProducer = new IngestionProducer(bucketdConfig, qpConfig,
-            s3Config);
+            s3Config, bucket);
+        this.bucket = bucket;
+        this.logger = logger;
+        this.logger.info('initializing raft log reader',
+            { method: 'IngestionReader.constructor',
+                bucketdConfig, raftId: this.raftId });
+        this.bucketdPrefix = bucketdConfig.prefix ? bucketdConfig.prefix :
```

Comment:
minor: `this.bucketdPrefix` has been removed in favor of `this._targetZenkoBucket` 
which is the zenko bucket name and should be unique

Reply:
Ok. @JianqinWang Please follow up with a PR for this.